plugins/source/gcp/resources/services/securitycenter/findings_fetch.go

```diff
@@ -2,28 +2,62 @@ package securitycenter
 
 import (
 	"context"
+	"fmt"
+	"time"
 
 	"google.golang.org/api/iterator"
 
 	pb "cloud.google.com/go/securitycenter/apiv1/securitycenterpb"
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugins/source/gcp/client"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	securitycenter "cloud.google.com/go/securitycenter/apiv1"
 )
 
-func fetchFindings(parent string) func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-	return func(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- any) error {
-		c := meta.(*client.Client)
-		req := &pb.ListFindingsRequest{
+const pageSize = 1000
+
+func getRequest(ctx context.Context, c *client.Client, table string, parent string) *pb.ListFindingsRequest {
+	filter, err := c.Backend.Get(ctx, table, c.ID())
+	if err != nil {
+		c.Logger().Warn().Str("table", table).Msgf("failed to get filter %s", err.Error())
+	}
+	var req *pb.ListFindingsRequest
+	if filter == "" || err != nil {
+		req = &pb.ListFindingsRequest{
 			Parent:   parent,
-			PageSize: 1000,
+			OrderBy:  "event_time",
+			PageSize: pageSize,
 		}
+	} else {
+		req = &pb.ListFindingsRequest{
+			Parent:   parent,
+			OrderBy:  "event_time",
+			PageSize: pageSize,
+			Filter:   filter,
+		}
+	}
+	return req
+}
+
+func setBackendState(ctx context.Context, c *client.Client, table string, lastEventTime *timestamppb.Timestamp) {
+	err := c.Backend.Set(ctx, table, c.ID(), fmt.Sprintf(`event_time >= "%s"`, lastEventTime.AsTime().Format(time.RFC3339)))
+	if err != nil {
+		c.Logger().Warn().Str("table", table).Msgf("failed to set filter %s", err.Error())
```

Comment:
Traditionally (in other plugins) we fail if any of the backend methods fail.

Reply:
I was wondering about that too. Should we fail? Not sure

Reply:
If it errored and ran without filter it would suddenly take too long and fetch it all (which might have other implications API-limits wise)... Leaving the decision to the user (either remove the state file or remove backend config) is better I think.

Reply:
Got it, makes sense. I'll return an error instead